### DISCUSSION TITLE
Adding the ~todo functionality

### DISCUSSION THREAD

**Discussion by theresnotime** [TheresNoGit/TheresNoDiscordBot#14] [Ideas]
In relation to #13, it would be a good idea to use something like a SQLite database to hold the tasks.

There's a fairly easy to use library ([sqlite3](https://docs.python.org/3/library/sqlite3.html)) with some pretty good [tutorials](https://www.sqlitetutorial.net/sqlite-python/) out there 😄

**Comment by theresnotime** [comment]
If you've not used SQL before, the below might be helpful in *creating* the SQLite table

```sql
CREATE TABLE "todo" (
	"user_id" INTEGER NOT NULL,
	"timestamp" TEXT NOT NULL,
	"task" TEXT NOT NULL,
	PRIMARY KEY ("user_id")
);
```

A word of caution, SQLite doesn't support the `DATETIME` type normally seen in MySQL.
[This article](https://www.sqlitetutorial.net/sqlite-date/) suggests some good ways of dealing with it

**Resources**
 - https://www.sqlite.org/datatype3.html
 - https://www.sqlite.org/lang_datefunc.html

**Reply by TamzinHadasa** [reply]
Yeah, personally I would have some constant called like `DT_FORMAT`, and store all dts formatted to that, and then turn them back into dts, using the same constant, when pulling from the database. That way the timestamps are readable in the database, unlike if you're using a Unix timestamp. But there's a case to be made for the latter too, especially if we plan to have anything other than Python access this data.

Also, do note: discord.py has a bug about converting dts to UTC rather than server time. (And UTC would be best.) Basically, **it gives the dt according to server time, but sets timezone as UTC**. This means you have to double-convert it to get the correct UTC time.

(This is at least last I checked. Possible they've fixed it. I can try to track down the relevant bug if necessary.)
